emit-ch4plume-v1/src/index.js

```diff
@@ -25,525 +25,772 @@ const markerProps = new Object();
+        const stopDate = new Date(
+          $("#slider-range").slider("values", 1) * 1000
+        );
+        const boolDisplay =
+          layerToggled && layerDate >= startDate && layerDate <= stopDate;
+        showHideLayers([layerID], boolDisplay);
+        $("#display_props").css({
+          visibility: boolDisplay ? "visible" : "hidden",
         });
-    }
-    onAdd(map) {
-        this.map = map;
-        this.container = document.createElement("div");
-        this.container.className = "mapboxgl-ctrl mapboxgl-ctrl-group";
-        this.container.addEventListener("contextmenu", (e) => e.preventDefault());
-        this.container.addEventListener("click", (e) => this.onClick());
-        this.container.innerHTML =
-            '<div class="tools-box">' +
-            "<button>" +
-            '<span id="layer-eye" class="mapboxgl-ctrl-icon btn fa fa-eye" aria-hidden="true" title="Show/Hide layers"></span>' +
-            "</button>" +
-            "</div>";
-        return this.container;
-    }
-    onRemove() {
-        this.container.parentNode.removeChild(this.container);
-        this.map = undefined;
-    }
+      } catch (error) {
+        console.error(error);
+      }
+    });
+  }
+  onAdd(map) {
+    this.map = map;
+    this.container = document.createElement("div");
+    this.container.className = "mapboxgl-ctrl mapboxgl-ctrl-group";
+    this.container.addEventListener("contextmenu", (e) => e.preventDefault());
+    this.container.addEventListener("click", (e) => this.onClick());
+    this.container.innerHTML =
+      '<div class="tools-box">' +
+      "<button>" +
+      '<span id="layer-eye" class="mapboxgl-ctrl-icon btn fa fa-eye" aria-hidden="true" title="Show/Hide layers"></span>' +
+      "</button>" +
+      "</div>";
+    return this.container;
+  }
+  onRemove() {
+    this.container.parentNode.removeChild(this.container);
+    this.map = undefined;
+  }
+}
+
+class MeasureDistance {
+  onClick() {
+    $("#measure-icon").toggleClass("measure-icon-clicked");
+    measureToggled = !measureToggled;
+    $("#display_props").toggleClass("hidden");
+  }
+  onAdd(map) {
+    this.map = map;
+    this.container = document.createElement("div");
+    this.container.className = "mapboxgl-ctrl mapboxgl-ctrl-group";
+    this.container.addEventListener("contextmenu", (e) => e.preventDefault());
+    this.container.addEventListener("click", (e) => this.onClick());
+    this.container.innerHTML =
+      '<div id="measure-icon-main">' +
+      "<button>" +
+      '<span id="measure-icon" class="mapboxgl-ctrl-icon" aria-hidden="true" title="Measure Tool">' +
+      '<i class="fa-solid fa-ruler-horizontal"></i>' +
+      "</span>" +
+      "</button>" +
+      "</div>";
+    return this.container;
+  }
+  onRemove() {
+    this.container.parentNode.removeChild(this.container);
+    this.map = undefined;
+  }
+}
+class ClearDistancePoints {
+  onClick() {
+    distancePoints.features.splice(0, distancePoints.features.length);
```

Comment:
Is this used to clear the features list? if yes we could rather do .length = 0 for a constant time operation.